supervisor/docker/interface.py

```diff
@@ -171,7 +179,10 @@ def _install(
                 self._docker_login(image)
 
             # Pull new image
-            docker_image = self.sys_docker.images.pull(f"{image}:{version!s}")
+            docker_image = self.sys_docker.images.pull(
+                f"{image}:{version!s}",
+                platform=MAP_ARCH[self.sys_arch.default],
```

Comment:
That will not work for add-ons. We have to add an parameter for the function and pass it from upper level function call

Reply:
i.e. https://github.com/home-assistant/supervisor/blob/main/supervisor/addons/model.py#L621-L622